packages/chips/ChipSet.js

```diff
@@ -29,27 +29,35 @@ import ChipCheckmark from './ChipCheckmark';
 
 export default class ChipSet extends Component {
   checkmarkWidth_ = 0;
-  foundation_ = null;
-  state = {
-    selectedChipIds: new Set(this.props.selectedChipIds),
-  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedChipIds: props.selectedChipIds,
```

Comment:
I think we don't need to keep track of `selectedChipIds` in state because it's already being kept track of within the foundation, and we should use that as the source of truth. More comments to explain below

Reply:
We need it on state because we need to be able to initialize the chips via props. Also if the user wants to update the Chips via props we need to be able to update the respective chips.